doc/developer/platform/reconciliation.md

```diff
@@ -0,0 +1,64 @@
+# Reconciliation of the ADAPTER/COMPUTE communication
+
+The current proposal for Materialize's platform includes the separation of the ADAPTER and COMPUTE components into distinct processes.
+Each represents its own failure domain and hence we need to design the communication protocols around failures and active replication (AR).
+In general, we need to define how commands and responses can be reconciled.
```

Comment:
Nit, but imo the right way to think of this is that the compute controller and the compute instances are in their own processes, and it is these two whose states need to be reconciled. The controller restarts, and wants to dictate to each instance what its state should be, and the instances (which may not have restarted) want to come up to that state, perhaps using a mechanism other than a reboot.

I think all of command reconciliation makes sense even without discussing ADAPTER.

Reply:
Further text indicates that this is already the thinking. Perhaps the text above is just stale.